src/coreclr/jit/lower.cpp

```diff
@@ -8179,8 +8185,354 @@ void Lowering::LowerIndir(GenTreeIndir* ind)
         const bool isContainable = false;
         TryCreateAddrMode(ind->Addr(), isContainable, ind);
     }
+
+#ifdef TARGET_ARM64
+    if (comp->opts.OptimizationEnabled() && ind->OperIs(GT_IND))
+    {
+        OptimizeForLdp(ind);
+    }
+#endif
+
+    return next;
 }
 
+#ifdef TARGET_ARM64
+
+//------------------------------------------------------------------------
+// OptimizeForLdp: Record information about an indirection, and try to optimize
+// it by moving it to be adjacent with a previous indirection such that they
+// can be transformed into 'ldp'.
+//
+// Arguments:
+//    ind - Indirection to record and to try to move.
+//
+// Returns:
+//    True if the optimization was successful.
+//
+bool Lowering::OptimizeForLdp(GenTreeIndir* ind)
+{
+    if (!ind->TypeIs(TYP_INT, TYP_LONG, TYP_DOUBLE, TYP_SIMD16))
+    {
+        return false;
+    }
```

Comment:
We need to fix #92766 to add `TYP_FLOAT` here.

Reply:
Are small types supported?

Reply:
Perhaps this needs to have `TYP_SIMD8` too? (on a rare occasion there is Vector64-based code for ARM64)

Reply:
> Perhaps this needs to have `TYP_SIMD8` too? (on a rare occasion there is Vector64-based code supprted by ARM64)

True, wondering if `TYP_SIMD8` was considered in this list?

Reply:
> Are small types supported?

Nope, there is no small variant of `ldp` I believe.

> Perhaps this needs to have TYP_SIMD8 too? (on a rare occasion there is Vector64-based code supprted by ARM64)
> 
> True, wondering if TYP_SIMD8 was considered in this list?

Yep, this can have `TYP_SIMD8` -- I forgot about those, thanks for spotting this.